libmamba/tests/src/specs/test_version_spec.cpp

```diff
@@ -352,6 +352,25 @@ namespace
             REQUIRE("~=3.3.2|==2.2"_vs.contains("3.3.3"_v));
             REQUIRE_FALSE("~=3.3.2|==2.2"_vs.contains("2.2.1"_v));
 
+            // Compatible version operator (~=)
+            REQUIRE("~=1.7"_vs.contains("1.7"_v));
+            REQUIRE("~=1.7"_vs.contains("1.7.0"_v));
+            REQUIRE("~=1.7"_vs.contains("1.7.1"_v));
+            REQUIRE_FALSE("~=1.7"_vs.contains("2.0.0"_v));
+            REQUIRE_FALSE("~=1.7"_vs.contains("1"_v));
+            REQUIRE_FALSE("~=1.7"_vs.contains("1.6.0"_v));
+            REQUIRE_FALSE("~=1.7"_vs.contains("0.1.0"_v));
+            REQUIRE("~=1.7.0"_vs.str() == "~=1.7.0");
```

Comment:
This is a first test the assumption described in https://github.com/mamba-org/mamba/pull/3651/files#r1952499642.